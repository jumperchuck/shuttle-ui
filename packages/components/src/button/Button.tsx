/**
 * Button
 */
import React from 'react';
import {
  ActivityIndicator,
  ActivityIndicatorProps,
  StyleProp,
  StyleSheet,
  TextStyle,
  TouchableOpacity,
  ViewStyle,
} from 'react-native';
import {
  ColorPropType,
  color as getColor,
  fontSize as getFontSize,
  ResponsiveValue,
} from '@shuttle-ui/system';
import { renderNode, RenderProps } from '@shuttle-ui/utils';
import { withColorMode } from '@shuttle-ui/color-mode';
import { withTheme } from '@shuttle-ui/theme';

import { ShuttleUIProps } from '../types';
import { Text, TextProps } from '../text';
import { Icon, IconProps } from '../icon';
import { Space, SpaceProps } from '../space/Space';

export interface ButtonProps extends SpaceProps {
  type?: 'solid' | 'text' | 'outline';
  color?: ColorPropType;
  size?: ResponsiveValue<'xs' | 'sm' | 'md' | 'lg'>;
  textStyle?: StyleProp<TextStyle>;
  disabledStyle?: StyleProp<ViewStyle>;
  disabledTextStyle?: StyleProp<TextStyle>;
  textProps?: TextProps;
  loading?: RenderProps<ActivityIndicatorProps>;
  leftLoading?: RenderProps<ActivityIndicatorProps>;
  rightLoading?: RenderProps<ActivityIndicatorProps>;
  icon?: RenderProps<IconProps>;
  leftIcon?: RenderProps<IconProps>;
  rightIcon?: RenderProps<IconProps>;
  children?: React.ReactNode;
}

export const Button = (props: ShuttleUIProps<ButtonProps>) => {
  const {
    type,
    color,
    size,
    style: styleProp,
    textStyle: textStyleProp,
    disabledStyle: disabledStyleProp,
    disabledTextStyle: disabledTextStyleProp,
    textProps: textPropsProp,
    loading,
    leftLoading,
    rightLoading,
    icon,
    leftIcon,
    rightIcon,
    onPress,
    disabled,
    children,
    theme,
    colorMode,
    ...rest
  } = props;

  type;
  size;

  const isLoading = loading || leftLoading || rightLoading;

  const spaceProps: ShuttleUIProps<SpaceProps> = {
    Component: TouchableOpacity,
    activeOpacity: 0.5,
    flexDirection: 'row',
    center: true,
    spacing: 10,
    borderColor: color,
    style: [styleProp, disabled && disabledStyleProp],
    onPress: isLoading ? onPress : undefined,
    disabled,
    theme,
    colorMode,
    ...rest,
  };

  const textStyle = StyleSheet.flatten([
    textStyleProp,
    textPropsProp?.style,
    disabled && disabledTextStyleProp,
  ]);

  const textProps: ShuttleUIProps<TextProps> = {
    color: 'text',
    theme,
    colorMode,
    ...textPropsProp,
    style: textStyle,
  };

  const fontSize = getFontSize({
    size: textStyle.fontSize ?? textProps.size ?? textProps.fontSize,
    theme,
    colorMode,
  });

  const textColor = getColor({
    color: textStyle.color ?? textProps.color,
    theme,
    colorMode,
  });

  const leftLoadingNode = loading || leftLoading;
  const rightLoadingNode = rightLoading;

  const leftIconNode = icon || leftIcon;
  const rightIconNode = rightIcon;

  const itemDefaultProps = {
    color: textColor,
    size: fontSize,
    theme,
    colorMode,
  };

  return (
    <Space {...spaceProps}>
      {leftLoadingNode &&
        renderNode(ActivityIndicator, leftLoadingNode, itemDefaultProps)}
      {!leftLoadingNode &&
        leftIconNode &&
        renderNode(
          Icon,
          typeof leftIconNode === 'string' ? { name: leftIconNode } : leftIconNode,
          itemDefaultProps,
        )}
      {renderNode(Text, children, textProps)}
      {!rightLoadingNode &&
        rightIconNode &&
        renderNode(
          Icon,
          typeof rightIconNode === 'string' ? { name: rightIconNode } : rightIconNode,
          itemDefaultProps,
        )}
      {rightLoadingNode &&
        renderNode(ActivityIndicator, rightLoadingNode, itemDefaultProps)}
    </Space>
  );
};

export default withColorMode(withTheme(Button, 'Button'));
