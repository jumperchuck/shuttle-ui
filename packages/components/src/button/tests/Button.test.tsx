import React from 'react';
import { render } from '@testing-library/react-native';

import Button from '../';

describe('Button Component', () => {
  it('should match snapshot', () => {
    const { toJSON } = render(<Button>Button</Button>);
    expect(toJSON()).toMatchSnapshot();
  });
});
