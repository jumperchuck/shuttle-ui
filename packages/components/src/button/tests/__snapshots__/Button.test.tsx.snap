// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should match snapshot 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#423F64",
      "borderColor": "#423F64",
      "borderWidth": 0,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 12,
      "paddingVertical": 8,
    }
  }
>
  <Image
    resizeMode="contain"
    style={
      Array [
        Object {},
        Array [
          Object {
            "height": 16,
            "tintColor": "#ffffff",
            "width": 16,
          },
          undefined,
        ],
      ]
    }
    theme={
      Object {
        "borders": Object {
          "1": Object {
            "borderColor": "#bdbdbd",
            "borderWidth": 0.5,
          },
          "2": Object {
            "borderColor": "#bdbdbd",
            "borderWidth": 1,
          },
          "3": Object {
            "borderColor": "#bdbdbd",
            "borderWidth": 2,
          },
        },
        "colors": Object {
          "accent": "#EE7A64",
          "amber": Object {
            "100": "#ffecb3",
            "200": "#ffe082",
            "300": "#ffd54f",
            "400": "#ffca28",
            "50": "#fff8e1",
            "500": "#ffc107",
            "600": "#ffb300",
            "700": "#ffa000",
            "800": "#ff8f00",
            "900": "#ff6f00",
            "A100": "#ffe57f",
            "A200": "#ffd740",
            "A400": "#ffc400",
            "A700": "#ffab00",
          },
          "background": "#ffffff",
          "black": "#000000",
          "blue": Object {
            "100": "#bbdefb",
            "200": "#90caf9",
            "300": "#64b5f6",
            "400": "#42a5f5",
            "50": "#e3f2fd",
            "500": "#2196f3",
            "600": "#1e88e5",
            "700": "#1976d2",
            "800": "#1565c0",
            "900": "#0d47a1",
            "A100": "#82b1ff",
            "A200": "#448aff",
            "A400": "#2979ff",
            "A700": "#2962ff",
          },
          "blueGrey": Object {
            "100": "#cfd8dc",
            "200": "#b0bec5",
            "300": "#90a4ae",
            "400": "#78909c",
            "50": "#eceff1",
            "500": "#607d8b",
            "600": "#546e7a",
            "700": "#455a64",
            "800": "#37474f",
            "900": "#263238",
          },
          "brown": Object {
            "100": "#d7ccc8",
            "200": "#bcaaa4",
            "300": "#a1887f",
            "400": "#8d6e63",
            "50": "#efebe9",
            "500": "#795548",
            "600": "#6d4c41",
            "700": "#5d4037",
            "800": "#4e342e",
            "900": "#3e2723",
          },
          "cyan": Object {
            "100": "#b2ebf2",
            "200": "#80deea",
            "300": "#4dd0e1",
            "400": "#26c6da",
            "50": "#e0f7fa",
            "500": "#00bcd4",
            "600": "#00acc1",
            "700": "#0097a7",
            "800": "#00838f",
            "900": "#006064",
            "A100": "#84ffff",
            "A200": "#18ffff",
            "A400": "#00e5ff",
            "A700": "#00b8d4",
          },
          "danger": "#c61700",
          "deepOrange": Object {
            "100": "#ffccbc",
            "200": "#ffab91",
            "300": "#ff8a65",
            "400": "#ff7043",
            "50": "#fbe9e7",
            "500": "#ff5722",
            "600": "#f4511e",
            "700": "#e64a19",
            "800": "#d84315",
            "900": "#bf360c",
            "A100": "#ff9e80",
            "A200": "#ff6e40",
            "A400": "#ff3d00",
            "A700": "#dd2c00",
          },
          "deepPurple": Object {
            "100": "#d1c4e9",
            "200": "#b39ddb",
            "300": "#9575cd",
            "400": "#7e57c2",
            "50": "#ede7f6",
            "500": "#673ab7",
            "600": "#5e35b1",
            "700": "#512da8",
            "800": "#4527a0",
            "900": "#311b92",
            "A100": "#b388ff",
            "A200": "#7c4dff",
            "A400": "#651fff",
            "A700": "#6200ea",
          },
          "disabled": "hsl(208, 8%, 90%)",
          "error": "#bf2c24",
          "foreground": "#ffffff",
          "green": Object {
            "100": "#c8e6c9",
            "200": "#a5d6a7",
            "300": "#81c784",
            "400": "#66bb6a",
            "50": "#e8f5e9",
            "500": "#4caf50",
            "600": "#43a047",
            "700": "#388e3c",
            "800": "#2e7d32",
            "900": "#1b5e20",
            "A100": "#b9f6ca",
            "A200": "#69f0ae",
            "A400": "#00e676",
            "A700": "#00c853",
          },
          "grey": Object {
            "100": "#f5f5f5",
            "200": "#eeeeee",
            "300": "#e0e0e0",
            "400": "#bdbdbd",
            "50": "#fafafa",
            "500": "#9e9e9e",
            "600": "#757575",
            "700": "#616161",
            "800": "#424242",
            "900": "#212121",
          },
          "indigo": Object {
            "100": "#c5cae9",
            "200": "#9fa8da",
            "300": "#7986cb",
            "400": "#5c6bc0",
            "50": "#e8eaf6",
            "500": "#3f51b5",
            "600": "#3949ab",
            "700": "#303f9f",
            "800": "#283593",
            "900": "#1a237e",
            "A100": "#8c9eff",
            "A200": "#536dfe",
            "A400": "#3d5afe",
            "A700": "#304ffe",
          },
          "lightBlue": Object {
            "100": "#b3e5fc",
            "200": "#81d4fa",
            "300": "#4fc3f7",
            "400": "#29b6f6",
            "50": "#e1f5fe",
            "500": "#03a9f4",
            "600": "#039be5",
            "700": "#0288d1",
            "800": "#0277bd",
            "900": "#01579b",
            "A100": "#80d8ff",
            "A200": "#40c4ff",
            "A400": "#00b0ff",
            "A700": "#0091ea",
          },
          "lightGreen": Object {
            "100": "#dcedc8",
            "200": "#c5e1a5",
            "300": "#aed581",
            "400": "#9ccc65",
            "50": "#f1f8e9",
            "500": "#8bc34a",
            "600": "#7cb342",
            "700": "#689f38",
            "800": "#558b2f",
            "900": "#33691e",
            "A100": "#ccff90",
            "A200": "#b2ff59",
            "A400": "#76ff03",
            "A700": "#64dd17",
          },
          "lime": Object {
            "100": "#f0f4c3",
            "200": "#e6ee9c",
            "300": "#dce775",
            "400": "#d4e157",
            "50": "#f9fbe7",
            "500": "#cddc39",
            "600": "#c0ca33",
            "700": "#afb42b",
            "800": "#9e9d24",
            "900": "#827717",
            "A100": "#f4ff81",
            "A200": "#eeff41",
            "A400": "#c6ff00",
            "A700": "#aeea00",
          },
          "orange": Object {
            "100": "#ffe0b2",
            "200": "#ffcc80",
            "300": "#ffb74d",
            "400": "#ffa726",
            "50": "#fff3e0",
            "500": "#ff9800",
            "600": "#fb8c00",
            "700": "#f57c00",
            "800": "#ef6c00",
            "900": "#e65100",
            "A100": "#ffd180",
            "A200": "#ffab40",
            "A400": "#ff9100",
            "A700": "#ff6d00",
          },
          "pink": Object {
            "100": "#f8bbd0",
            "200": "#f48fb1",
            "300": "#f06292",
            "400": "#ec407a",
            "50": "#fce4ec",
            "500": "#e91e63",
            "600": "#d81b60",
            "700": "#c2185b",
            "800": "#ad1457",
            "900": "#880e4f",
            "A100": "#ff80ab",
            "A200": "#ff4081",
            "A400": "#f50057",
            "A700": "#c51162",
          },
          "primary": "#423F64",
          "purple": Object {
            "100": "#e1bee7",
            "200": "#ce93d8",
            "300": "#ba68c8",
            "400": "#ab47bc",
            "50": "#f3e5f5",
            "500": "#9c27b0",
            "600": "#8e24aa",
            "700": "#7b1fa2",
            "800": "#6a1b9a",
            "900": "#4a148c",
            "A100": "#ea80fc",
            "A200": "#e040fb",
            "A400": "#d500f9",
            "A700": "#aa00ff",
          },
          "red": Object {
            "100": "#ffcdd2",
            "200": "#ef9a9a",
            "300": "#e57373",
            "400": "#ef5350",
            "50": "#ffebee",
            "500": "#f44336",
            "600": "#e53935",
            "700": "#d32f2f",
            "800": "#c62828",
            "900": "#b71c1c",
            "A100": "#ff8a80",
            "A200": "#ff5252",
            "A400": "#ff1744",
            "A700": "#d50000",
          },
          "secondary": "#2C2A4B",
          "success": "#439946",
          "surface": "#ffffff",
          "teal": Object {
            "100": "#b2dfdb",
            "200": "#80cbc4",
            "300": "#4db6ac",
            "400": "#26a69a",
            "50": "#e0f2f1",
            "500": "#009688",
            "600": "#00897b",
            "700": "#00796b",
            "800": "#00695c",
            "900": "#004d40",
            "A100": "#a7ffeb",
            "A200": "#64ffda",
            "A400": "#1de9b6",
            "A700": "#00bfa5",
          },
          "text": "#000000",
          "warning": "#cfbe27",
          "white": "#ffffff",
          "yellow": Object {
            "100": "#fff9c4",
            "200": "#fff59d",
            "300": "#fff176",
            "400": "#ffee58",
            "50": "#fffde7",
            "500": "#ffeb3b",
            "600": "#fdd835",
            "700": "#fbc02d",
            "800": "#f9a825",
            "900": "#f57f17",
            "A100": "#ffff8d",
            "A200": "#ffff00",
            "A400": "#ffea00",
            "A700": "#ffd600",
          },
        },
        "fontFamilies": Object {},
        "fontSizes": Object {
          "lg": 18,
          "md": 16,
          "sm": 14,
          "xl": 20,
          "xs": 12,
          "xxl": 24,
          "xxs": 8,
        },
        "fontVariants": Object {},
        "fontWeights": Object {
          "bold": "bold",
          "light": "normal",
          "normal": "normal",
        },
        "props": Object {},
        "radius": Object {
          "lg": 20,
          "md": 12,
          "sm": 8,
          "xl": 30,
          "xs": 4,
          "xxl": 40,
          "xxs": 2,
        },
        "shadows": Object {
          "1": Object {
            "elevation": 1,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0.24,
            "shadowRadius": 0.75,
          },
          "2": Object {
            "elevation": 2,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 0.5,
              "width": 0,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 1.5,
          },
          "3": Object {
            "elevation": 3,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 1,
              "width": 1,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 2,
          },
        },
        "spacings": Object {
          "lg": 20,
          "md": 12,
          "sm": 8,
          "xl": 30,
          "xs": 4,
          "xxl": 40,
          "xxs": 2,
        },
      }
    }
  />
  <Text
    style={
      Array [
        Object {},
        Object {
          "color": "white",
          "marginLeft": 10,
          "textAlign": "center",
        },
      ]
    }
    theme={
      Object {
        "borders": Object {
          "1": Object {
            "borderColor": "#bdbdbd",
            "borderWidth": 0.5,
          },
          "2": Object {
            "borderColor": "#bdbdbd",
            "borderWidth": 1,
          },
          "3": Object {
            "borderColor": "#bdbdbd",
            "borderWidth": 2,
          },
        },
        "colors": Object {
          "accent": "#EE7A64",
          "amber": Object {
            "100": "#ffecb3",
            "200": "#ffe082",
            "300": "#ffd54f",
            "400": "#ffca28",
            "50": "#fff8e1",
            "500": "#ffc107",
            "600": "#ffb300",
            "700": "#ffa000",
            "800": "#ff8f00",
            "900": "#ff6f00",
            "A100": "#ffe57f",
            "A200": "#ffd740",
            "A400": "#ffc400",
            "A700": "#ffab00",
          },
          "background": "#ffffff",
          "black": "#000000",
          "blue": Object {
            "100": "#bbdefb",
            "200": "#90caf9",
            "300": "#64b5f6",
            "400": "#42a5f5",
            "50": "#e3f2fd",
            "500": "#2196f3",
            "600": "#1e88e5",
            "700": "#1976d2",
            "800": "#1565c0",
            "900": "#0d47a1",
            "A100": "#82b1ff",
            "A200": "#448aff",
            "A400": "#2979ff",
            "A700": "#2962ff",
          },
          "blueGrey": Object {
            "100": "#cfd8dc",
            "200": "#b0bec5",
            "300": "#90a4ae",
            "400": "#78909c",
            "50": "#eceff1",
            "500": "#607d8b",
            "600": "#546e7a",
            "700": "#455a64",
            "800": "#37474f",
            "900": "#263238",
          },
          "brown": Object {
            "100": "#d7ccc8",
            "200": "#bcaaa4",
            "300": "#a1887f",
            "400": "#8d6e63",
            "50": "#efebe9",
            "500": "#795548",
            "600": "#6d4c41",
            "700": "#5d4037",
            "800": "#4e342e",
            "900": "#3e2723",
          },
          "cyan": Object {
            "100": "#b2ebf2",
            "200": "#80deea",
            "300": "#4dd0e1",
            "400": "#26c6da",
            "50": "#e0f7fa",
            "500": "#00bcd4",
            "600": "#00acc1",
            "700": "#0097a7",
            "800": "#00838f",
            "900": "#006064",
            "A100": "#84ffff",
            "A200": "#18ffff",
            "A400": "#00e5ff",
            "A700": "#00b8d4",
          },
          "danger": "#c61700",
          "deepOrange": Object {
            "100": "#ffccbc",
            "200": "#ffab91",
            "300": "#ff8a65",
            "400": "#ff7043",
            "50": "#fbe9e7",
            "500": "#ff5722",
            "600": "#f4511e",
            "700": "#e64a19",
            "800": "#d84315",
            "900": "#bf360c",
            "A100": "#ff9e80",
            "A200": "#ff6e40",
            "A400": "#ff3d00",
            "A700": "#dd2c00",
          },
          "deepPurple": Object {
            "100": "#d1c4e9",
            "200": "#b39ddb",
            "300": "#9575cd",
            "400": "#7e57c2",
            "50": "#ede7f6",
            "500": "#673ab7",
            "600": "#5e35b1",
            "700": "#512da8",
            "800": "#4527a0",
            "900": "#311b92",
            "A100": "#b388ff",
            "A200": "#7c4dff",
            "A400": "#651fff",
            "A700": "#6200ea",
          },
          "disabled": "hsl(208, 8%, 90%)",
          "error": "#bf2c24",
          "foreground": "#ffffff",
          "green": Object {
            "100": "#c8e6c9",
            "200": "#a5d6a7",
            "300": "#81c784",
            "400": "#66bb6a",
            "50": "#e8f5e9",
            "500": "#4caf50",
            "600": "#43a047",
            "700": "#388e3c",
            "800": "#2e7d32",
            "900": "#1b5e20",
            "A100": "#b9f6ca",
            "A200": "#69f0ae",
            "A400": "#00e676",
            "A700": "#00c853",
          },
          "grey": Object {
            "100": "#f5f5f5",
            "200": "#eeeeee",
            "300": "#e0e0e0",
            "400": "#bdbdbd",
            "50": "#fafafa",
            "500": "#9e9e9e",
            "600": "#757575",
            "700": "#616161",
            "800": "#424242",
            "900": "#212121",
          },
          "indigo": Object {
            "100": "#c5cae9",
            "200": "#9fa8da",
            "300": "#7986cb",
            "400": "#5c6bc0",
            "50": "#e8eaf6",
            "500": "#3f51b5",
            "600": "#3949ab",
            "700": "#303f9f",
            "800": "#283593",
            "900": "#1a237e",
            "A100": "#8c9eff",
            "A200": "#536dfe",
            "A400": "#3d5afe",
            "A700": "#304ffe",
          },
          "lightBlue": Object {
            "100": "#b3e5fc",
            "200": "#81d4fa",
            "300": "#4fc3f7",
            "400": "#29b6f6",
            "50": "#e1f5fe",
            "500": "#03a9f4",
            "600": "#039be5",
            "700": "#0288d1",
            "800": "#0277bd",
            "900": "#01579b",
            "A100": "#80d8ff",
            "A200": "#40c4ff",
            "A400": "#00b0ff",
            "A700": "#0091ea",
          },
          "lightGreen": Object {
            "100": "#dcedc8",
            "200": "#c5e1a5",
            "300": "#aed581",
            "400": "#9ccc65",
            "50": "#f1f8e9",
            "500": "#8bc34a",
            "600": "#7cb342",
            "700": "#689f38",
            "800": "#558b2f",
            "900": "#33691e",
            "A100": "#ccff90",
            "A200": "#b2ff59",
            "A400": "#76ff03",
            "A700": "#64dd17",
          },
          "lime": Object {
            "100": "#f0f4c3",
            "200": "#e6ee9c",
            "300": "#dce775",
            "400": "#d4e157",
            "50": "#f9fbe7",
            "500": "#cddc39",
            "600": "#c0ca33",
            "700": "#afb42b",
            "800": "#9e9d24",
            "900": "#827717",
            "A100": "#f4ff81",
            "A200": "#eeff41",
            "A400": "#c6ff00",
            "A700": "#aeea00",
          },
          "orange": Object {
            "100": "#ffe0b2",
            "200": "#ffcc80",
            "300": "#ffb74d",
            "400": "#ffa726",
            "50": "#fff3e0",
            "500": "#ff9800",
            "600": "#fb8c00",
            "700": "#f57c00",
            "800": "#ef6c00",
            "900": "#e65100",
            "A100": "#ffd180",
            "A200": "#ffab40",
            "A400": "#ff9100",
            "A700": "#ff6d00",
          },
          "pink": Object {
            "100": "#f8bbd0",
            "200": "#f48fb1",
            "300": "#f06292",
            "400": "#ec407a",
            "50": "#fce4ec",
            "500": "#e91e63",
            "600": "#d81b60",
            "700": "#c2185b",
            "800": "#ad1457",
            "900": "#880e4f",
            "A100": "#ff80ab",
            "A200": "#ff4081",
            "A400": "#f50057",
            "A700": "#c51162",
          },
          "primary": "#423F64",
          "purple": Object {
            "100": "#e1bee7",
            "200": "#ce93d8",
            "300": "#ba68c8",
            "400": "#ab47bc",
            "50": "#f3e5f5",
            "500": "#9c27b0",
            "600": "#8e24aa",
            "700": "#7b1fa2",
            "800": "#6a1b9a",
            "900": "#4a148c",
            "A100": "#ea80fc",
            "A200": "#e040fb",
            "A400": "#d500f9",
            "A700": "#aa00ff",
          },
          "red": Object {
            "100": "#ffcdd2",
            "200": "#ef9a9a",
            "300": "#e57373",
            "400": "#ef5350",
            "50": "#ffebee",
            "500": "#f44336",
            "600": "#e53935",
            "700": "#d32f2f",
            "800": "#c62828",
            "900": "#b71c1c",
            "A100": "#ff8a80",
            "A200": "#ff5252",
            "A400": "#ff1744",
            "A700": "#d50000",
          },
          "secondary": "#2C2A4B",
          "success": "#439946",
          "surface": "#ffffff",
          "teal": Object {
            "100": "#b2dfdb",
            "200": "#80cbc4",
            "300": "#4db6ac",
            "400": "#26a69a",
            "50": "#e0f2f1",
            "500": "#009688",
            "600": "#00897b",
            "700": "#00796b",
            "800": "#00695c",
            "900": "#004d40",
            "A100": "#a7ffeb",
            "A200": "#64ffda",
            "A400": "#1de9b6",
            "A700": "#00bfa5",
          },
          "text": "#000000",
          "warning": "#cfbe27",
          "white": "#ffffff",
          "yellow": Object {
            "100": "#fff9c4",
            "200": "#fff59d",
            "300": "#fff176",
            "400": "#ffee58",
            "50": "#fffde7",
            "500": "#ffeb3b",
            "600": "#fdd835",
            "700": "#fbc02d",
            "800": "#f9a825",
            "900": "#f57f17",
            "A100": "#ffff8d",
            "A200": "#ffff00",
            "A400": "#ffea00",
            "A700": "#ffd600",
          },
        },
        "fontFamilies": Object {},
        "fontSizes": Object {
          "lg": 18,
          "md": 16,
          "sm": 14,
          "xl": 20,
          "xs": 12,
          "xxl": 24,
          "xxs": 8,
        },
        "fontVariants": Object {},
        "fontWeights": Object {
          "bold": "bold",
          "light": "normal",
          "normal": "normal",
        },
        "props": Object {},
        "radius": Object {
          "lg": 20,
          "md": 12,
          "sm": 8,
          "xl": 30,
          "xs": 4,
          "xxl": 40,
          "xxs": 2,
        },
        "shadows": Object {
          "1": Object {
            "elevation": 1,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0.24,
            "shadowRadius": 0.75,
          },
          "2": Object {
            "elevation": 2,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 0.5,
              "width": 0,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 1.5,
          },
          "3": Object {
            "elevation": 3,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 1,
              "width": 1,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 2,
          },
        },
        "spacings": Object {
          "lg": 20,
          "md": 12,
          "sm": 8,
          "xl": 30,
          "xs": 4,
          "xxl": 40,
          "xxs": 2,
        },
      }
    }
  >
    Button
  </Text>
</View>
`;
